pandas/io/formats/format.py

```diff
@@ -1103,7 +1103,11 @@ def format_array(
     List[str]
     """
     fmt_klass: type[_GenericArrayFormatter]
-    if lib.is_np_dtype(values.dtype, "M"):
+    if hasattr(values, "type") and values.type == "null":
```

Comment:
can we do something more explicit than hasattr checks?  i.e. isinstance(dtype, ListDtype) or whatever?